index.bs

```diff
@@ -686,16 +686,71 @@ dictionary MLOperandDescriptor {
 
 ### The MLOperand interface ### {#api-mloperand}
 
-An {{MLOperand}} represents an intermediary graph being constructed as a result of compositing parts of an operation into a fully composed operation.
+An {{MLOperand}} represents an intermediary graph edge being constructed as a result of compositing parts of an operation into a fully composed operation.
 
 For instance, an {{MLOperand}} may represent a constant feeding to an operation or the result from combining multiple constants together into an operation. See also [[#programming-model]].
 
 <script type=idl>
+enum MLOperandKind {"constant", "input", "output"};
 [SecureContext, Exposed=(Window, DedicatedWorker)]
 interface MLOperand {};
 </script>
 
-See also [[#security-new-ops]]
+{{MLOperand}} has the following internal slots:
+<div class=internal-slots>
+<dl dfn-type=attribute dfn-for="MLOperand">
+    : <dfn>\[[builder]]</dfn> of type {{MLGraphBuilder}}
+    ::
+        The {{MLOperand}}'s associated builder object.
+
+    : <dfn>\[[kind]]</dfn> of type {{MLOperandKind}}
+    ::
+        The {{MLOperand}}'s kind.
+
+    : <dfn>\[[descriptor]]</dfn> of type {{MLOperandDescriptor}}
+    ::
+        The {{MLOperand}}'s descriptor.
+
+    : <dfn>\[[name]]</dfn> of type [=string=]
+    ::
+        The {{MLOperand}}'s name (only for input operands).
+
+    : <dfn>\[[operand]]</dfn> of type [=object=]
+    ::
+        Reference to {{MLOperand}}'s corresponding [=implementation-defined=] platform operand object.
+
+    : <dfn>\[[operator]]</dfn> of type [=object=]
+    ::
+        Reference to {{MLOperand}}'s corresponding [=implementation-defined=] platform operator object.
+</dl>
+</div>
+
+Since the {{MLOperand/[[builder]]}} object is bound by the {{MLGraphBuilder/constructor()}} constructor to an {{MLContext}} object, an {{MLOperand}} is also always bound to the same {{MLContext}} object.
+
+#### Creating {{MLOperand}} #### {#api-mloperand-create}
+<div class="note">
+    The {{MLOperand}} objects are created internally by the methods of {{MLGraphBuilder}}. Based on these steps, exposing a constructor would be possible as well, but needs an explicit use case.
```

Comment:
Why? `MLOperand` is an interface that represents either an individual operation or the whole composition of operations being constructed by the graph builder. Besides the methods to create the operands for graph inputs, those methods off the graph builder should already represent every operation as `MLOperand`. There should be no need for additional explicit constructor of `MLOperand` if that's what you mean.

Reply:
OK, will remove the note.

Reply:
The reasoning behind constructors usually is that APIs should be designed in a way that it's easy to create JS objects from console/scripts that can be connected via their interfaces to the underlying representations.

If we want to keep a tight connection between the JS objects and what they represent, then we can stick with the factory-only design, and use default object constructor steps in their algorithms.

Reply:
We consciously chose the builder method of construction when we define the graph builder interface. More specifically for the work to construct the ML graph, the builder pattern works much better as it allows the implementation to keep transient graph building state in the graph builder interface as the construction progresses. In short, `MLOperand` isn't designed with a constructor pattern in mind.